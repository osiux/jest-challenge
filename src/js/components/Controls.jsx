import React from 'react';
import PropTypes from 'prop-types';

import Button from './Button';

const Controls = ({ onDecrease, onReset, onIncrease }) => (
    <div className="card-body ">
        <div className="btn-group btn-group-justified">
            <Button onClick={onDecrease}>-</Button>
            <Button context="danger" onClick={onReset}>
                Reset
            </Button>
            <Button onClick={onIncrease}>+</Button>
        </div>
    </div>
);

Controls.propTypes = {
    onDecrease: PropTypes.func.isRequired,
    onReset: PropTypes.func.isRequired,
    onIncrease: PropTypes.func.isRequired
};

export default Controls;
