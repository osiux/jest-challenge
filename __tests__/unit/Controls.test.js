import React from 'react';
import { shallow } from 'enzyme';

import Controls from '../../src/js/components/Controls';

describe('Controls component', () => {
    it('matches snapshot', () => {
        const mockFn = jest.fn();

        const component = shallow(
            <Controls
                onDecrease={mockFn}
                onReset={mockFn}
                onIncrease={mockFn}
            />
        );

        expect(component).toMatchSnapshot();
    });

    it('renders control Buttons', () => {
        const mockFn = jest.fn();

        const component = shallow(
            <Controls
                onDecrease={mockFn}
                onReset={mockFn}
                onIncrease={mockFn}
            />
        );

        const buttons = component.find('Button');

        expect(buttons).toHaveLength(3);
        expect(buttons.filter('[context="primary"]')).toHaveLength(2);
        expect(buttons.filter('[context="danger"]')).toHaveLength(1);
    });
});
