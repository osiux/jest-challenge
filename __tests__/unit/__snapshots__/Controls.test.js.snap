// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component matches snapshot 1`] = `
<div
  className="card-body "
>
  <div
    className="btn-group btn-group-justified"
  >
    <Button
      context="primary"
      onClick={[MockFunction]}
    >
      -
    </Button>
    <Button
      context="danger"
      onClick={[MockFunction]}
    >
      Reset
    </Button>
    <Button
      context="primary"
      onClick={[MockFunction]}
    >
      +
    </Button>
  </div>
</div>
`;
